controller/cmd/service-mirror/cluster_watcher.go

```diff
@@ -1121,9 +1148,16 @@ func (rcsw *RemoteClusterServiceWatcher) extractGatewaySpec(gateway *corev1.Serv
 
 	var gatewayEndpoints []corev1.EndpointAddress
 	for _, ingress := range gateway.Status.LoadBalancer.Ingress {
+		ip := ingress.IP
+		if ip == "" {
+			ipAddr, err := net.ResolveIPAddr("ip", ingress.Hostname)
```

Comment:
Just to verify. What happens here if this name resolution returns an IPv6 address and there is no dual stack feature enabled on this cluster. Would that still work ? If not, should we make sure this is not an IPv6 addr ?

Reply:
To be honest, I have no idea what the state of our ipv6 support is in Linkerd.

Reply:
Are we actually making IPv4-specific assumptions here? We need to document this at the very least. It's debt.

Reply:
I don't think there are any ipv4-specific assumptions here.  `ResolveIPAddr` can return either ipv4 or ipv6 addresses and the `String()` implementation will render them correctly.  So we support ipv6 here to whatever extent that kubernetes does.